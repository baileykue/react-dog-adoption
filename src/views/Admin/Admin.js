import React from 'react';
import DogForm from '../../components/DogForm/DogForm';

export default function Admin() {
  return (
    <div>
      <DogForm />
    </div>
  );
}
