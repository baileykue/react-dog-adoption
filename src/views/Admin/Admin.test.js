import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Admin from './Admin';

test('<Admin/> should render the empty dog form', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/admin']}>
      <Route path="/dogs/admin">
        <Admin />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Breed:');
  expect(container).toMatchSnapshot();
});
