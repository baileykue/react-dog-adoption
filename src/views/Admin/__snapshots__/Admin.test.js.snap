// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin/> should render the empty dog form 1`] = `
<div>
  <div>
    <div>
      <form
        class="form"
      >
        <label>
          Name: 
        </label>
        <input
          placeholder="Name"
          type="text"
          value=""
        />
        <label>
          Bio: 
        </label>
        <textarea
          placeholder="Bio"
          type="text"
        />
        <label>
          Image: 
        </label>
        <input
          placeholder="Image"
          type="text"
          value=""
        />
        <label>
          Age: 
        </label>
        <input
          placeholder="Age"
          type="text"
          value=""
        />
        <label>
          Breed: 
        </label>
        <input
          placeholder="Breed"
          type="text"
          value=""
        />
        <button>
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
